cmake/kokkos_functions.cmake

```diff
@@ -982,6 +982,13 @@ function(kokkos_check_flags)
     include(CheckCompilerFlag)
     #delete cache so we always do the check
     unset(KOKKOS_COMPILE_OPTIONS_CHECK CACHE)
+    if(INP_LINKER_FLAGS)
+      #check_compiler/linker_flag requires a whitespace separated list
+      string(REPLACE ";" " " WHITESPACE_LINKER_FLAGS "${INP_LINKER_FLAGS}")
+      # icpx adds "-device ..." options that need quotes (which CMake removes). We need to add them here again.
+      string(REGEX REPLACE "(-device [A-Za-z0-9_\\\\.]*)" "\"\\1\"" QUOTED_LINKER_FLAGS "${WHITESPACE_LINKER_FLAGS}")
```

Comment:
@pzehner it might be that we need this fumbling to make the quoted device flags with SYCL work ... we had a similar issue with the compiler flags